src/server/run-websocket-server.ts

```diff
@@ -1,29 +1,54 @@
 import * as WebSocket from "ws";
 import * as Knex from "knex";
+import { EventEmitter } from "events";
 import { JsonRpcRequest } from "../types";
 import { isJsonRpcRequest } from "./is-json-rpc-request";
 import { dispatchJsonRpcRequest } from "./dispatch-json-rpc-request";
 import { makeJsonRpcResponse } from "./make-json-rpc-response";
+import { Subscriptions } from "./subscriptions";
 
 export function runWebsocketServer(db: Knex, port: number): void {
   const websocketServer: WebSocket.Server = new WebSocket.Server({ port });
   websocketServer.on("connection", (websocket: WebSocket): void => {
+    const subscriptions = new Subscriptions();
+
     websocket.on("message", (data: WebSocket.Data): void => {
       try {
         const message = JSON.parse(data as string);
-        if (isJsonRpcRequest(message)) {
+        if (!isJsonRpcRequest(message)) return console.error("bad json rpc message received:", message);
+
+        if (message.method === "subscribe") {
+          const eventName: string = message.params.shift();
+
+          try {
+            const subscription: string = subscriptions.subscribe(eventName, message.params, (data: {}): void => {
+              websocket.send(makeJsonRpcResponse(message.id, { subscription, result: data}));
+            });
+            websocket.send(makeJsonRpcResponse(message.id, { subscription }));
+          } catch (exc) {
+            console.error("suscription error", exc, data);
+            websocket.send(makeJsonRpcResponse(message.id, false));
```

Comment:
We should probably have a separate `makeJsonRpcError` function that creates the error response in the format of http://www.jsonrpc.org/specification#error_object

Reply:
Good call!

Reply:
I'll take this chance to translate all these errors to the spec.